### PR TITLE
Extend image service

### PR DESCRIPTION
**Pull request by bbx-winner**
- feat: `tap_cv` action supports ui type detection and tap
- feat: support auto close popups button detection and tap by indicating `close_popup` in testcase config
- feat: support scenario(`ui_live_type`) validation
- fix: wait 500 ms for last swipe done before GetScreenResult especially for iOS devices
- change: ocr texts default validation method(when regex off): equal -> contains
- compatibility: add default `max_retry_times` for `swipe_to_tap_app` action
- compatibility: support indicating `options` separately in `actions` level
- compatibility: `index` starts from 1 if index > 0

### CODE REVIEW

File: hrp/pkg/uixt/android_device.go [hunk 1]
```diff
@@ -156,7 +157,8 @@ func (dev *AndroidDevice) LogEnabled() bool {
 
 func (dev *AndroidDevice) NewDriver(capabilities Capabilities) (driverExt *DriverExt, err error) {
 	var driver WebDriver
-	if dev.UIA2 {
+	envDisableUIA2 := env.DISABLE_UIAUTOMATOR_SERVER == "true"
+	if dev.UIA2 || !envDisableUIA2 {
```
Comment (bbx-winner) [inline hrp/pkg/uixt/android_device.go:171]: TODO：采用更合理方式指定 adb/uia2 驱动

File: hrp/pkg/uixt/android_device.go [hunk 2]
```diff
@@ -145,6 +154,7 @@ type AndroidDevice struct {
 	UIA2IP       string `json:"uia2_ip,omitempty" yaml:"uia2_ip,omitempty"`     // uiautomator2 server ip
 	UIA2Port     int    `json:"uia2_port,omitempty" yaml:"uia2_port,omitempty"` // uiautomator2 server port
 	LogOn        bool   `json:"log_on,omitempty" yaml:"log_on,omitempty"`
+	ClosePopup   bool   `json:"close_popup,omitempty" yaml:"close_popup,omitempty"`
```
Comment (bbx-winner) [inline hrp/pkg/uixt/android_device.go:157]: 全局（隐式）弹窗处理：config 中的 close_popup 开启/关闭
显示调用弹窗处理：tap_ocr/ tap_cv

File: hrp/pkg/uixt/action.go [hunk 3]
```diff
@@ -35,6 +35,7 @@ const (
 	SelectorOCR           string = "ui_ocr"
 	SelectorImage         string = "ui_image"
 	SelectorForegroundApp string = "ui_foreground_app"
+	SelectorLiveType      string = "ui_live_type"
```
Comment (bbx-winner) [inline hrp/pkg/uixt/action.go:38]: 场景识别可以作为标注，而不是断言的形式存在

File: docs/CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,18 @@
...
 
+## v4.3.5 (2023-07-17)
+
+**go version**
+
+- feat: `tap_cv` action supports ui type detection and tap
+- feat: support auto close popups button detection and tap by indicating `close_popup` in testcase config
+- feat: support scenario(`ui_live_type`) validation
+- fix: wait 500 ms for last swipe done before GetScreenResult especially for iOS devices
+- change: ocr texts default validation method(when regex off): equal -> contains
```
Comment (debugtalk) [inline docs/CHANGELOG.md:11]: 不合理

File: hrp/pkg/uixt/ocr_vedem.go [hunk 5]
```diff
@@ -94,7 +96,7 @@ func (t OCRTexts) FindText(text string, options ...ActionOption) (
 			}
 		} else {
 			// regex off, check if match exactly
-			if ocrText.Text != text {
+			if !strings.Contains(ocrText.Text, text) {
```
Comment (bbx-winner) [inline hrp/pkg/uixt/ocr_vedem.go:99]: 默认还是采用 == 来进行判断，response.go 中如果是 exists 则走 contains 逻辑

**Pull request closed without merging**